### PR TITLE
Feature/orm stack part2 - mushop-stack-v2.0.0.zip release

### PR DESCRIPTION
**Pull request by junior**
Includes updates to allow deploy everything from the stack, including ATP.

Also include the updated schema for ORM deployment. Code changes to allow use local terraform, oracle resource manager, terraform cloud seamlessly.

"Gold Master" stack included here for testing: 
[mushop-stack-v2.0.0.GM.zip](https://github.com/oracle-quickstart/oci-cloudnative/files/4659555/mushop-stack-v2.0.0.GM.zip)

### CODE REVIEW

File: deploy/complete/helm-chart/setup/values.yaml [hunk 1]
```diff
@@ -10,6 +10,13 @@ metrics-server:
 # https://github.com/helm/charts/blob/master/stable/prometheus/README.md
 prometheus:
   enabled: true
+  extraScrapeConfigs: |
+    - job_name: 'ingress-nginx'
+      metrics_path: /metrics
+      scrape_interval: 5s
+      static_configs:
+        - targets:
+          - mushop-utils-ingress-nginx-controller-metrics:9913
```
Comment (mvandervliet) [inline deploy/complete/helm-chart/setup/values.yaml:19]: is it possible to set this target dynamically? I _really really_ dislike having any hard-coded references to the namespace in which this is installed

File: deploy/complete/terraform/mushop.tf [hunk 2]
```diff
@@ -17,9 +17,30 @@ resource "helm_release" "mushop" {
...
   wait      = false
 
-  values = [
-    file("${path.module}/../helm-chart/mushop/values-mock.yaml"),
-  ]
+  set_string {
+    name  = "global.mock.service"
+    value = var.mushop_mock_mode_all ? "all" : "false"
+  }
+  set {
```
Comment (mvandervliet) [inline deploy/complete/terraform/mushop.tf:24]: I don't think these conditional settings are necessary. If mock mode is `"all"` then these values are essentially ignored

File: deploy/complete/terraform/schema.yaml [hunk 3]
```diff
@@ -7,12 +7,14 @@ description: "Oracle Container Engine (OKE) stack for MuShop App"
...
...18fkBn6Y5K3i+wl2oWpDE+LQXiXChFzCd8tNXMNsP78Xs96/hce3HSI8t45o2c3ewXF5Bq/UoJ8FPkL6PsD9iJ5DhNm9JhHsTuXbgN/SZ49iHod7kfYUvdouItjHJ81UdSF3q0SYwCwW3SmXsYd5UPYlT4JnYUVECBk9u3vPqIbbMU9yPygSvJHsjSP7GRtOv3SouQl2N+Fa4DUyDhcVKDnP3lgbGBcRBs54Tu0FiTCBeQjmKYm7xzBFN7wkSMeW0z/hXlANISYD+EYZPReAT1AjIXOfwC0paJ8g11oi2FTuOxQU6WEih5/zKqEQ4s/IaC0R3MyiScwq4iPUzMPzaB8RpjGFJydl/Pwb9bJ7PVroNUxKGrxW7blbJPAG4j6TCPOYCKKtXvKANxBLj6FbLabfRiKEmFDyK3UTnyG70ITHs2H3VdoamWttJMKcYgbzcoMeYGefrtpWIkw4ruRqG4lwDFNFBKkDv6pYHlOM6l8Pq3paO0kE+/ALG0V8iOzaxR7JmJFEsPbBRtuIMAf8DGZhaUNqwZOgOhHij0lqDRGulUToYzagnPdEqIQjmDjMGjWe6bSTRHgxpmLICjWeReTBT0sqrGJK9bSKCLOMKpD1MBVC/JpCedjNPgckERbbRoTjjApe9jD7FD3Kwz7VpdNWIpxk9PS3JdpdIn+nEGhCueV5lttEhAD4OTF5GfOAD68WquE6Rlv/v0ONPR47QYRQEmFa0mDBj2flsTujflyo24/BDjT+hBofYnbpeCJUm0yHpRom2kiEEJOU2sUEkb5Fu5+cspPewqsZFeJ+oo6huFMS4Wd1Az1GFT88ymECs3TfwRQFuUzNNZqdIIItTL1MxZUyTwKukWQNMSl9i3UNbk+E9uEg8CrMquM6Zjv8lSYGZjtJMIGJhHkiVLex5jEPKAez72OBBsoEbScROhJptgycNxTLYw5TK+K4/v8IDQXjwm1ms638ZZ9n5NPVy03a64G3MHpu9Geb6sPtJEJfouzPMU8iu4CPKJbBPPAixQ82MSUCGqsXuZ1EGDJ6gPYmPjexLAbA9zHR2IuYRN/GXO/nbPPNPIOphPZ9fP5BWfwQ+D/gm5itgF9Wf3p4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHh4eHi0Ff8PIS+RFSylw7QAAAAASUVORK5CYII=
```
Comment (mvandervliet) [inline deploy/complete/terraform/schema.yaml:10]: I'll work on `svg` for you

File: deploy/complete/helm-chart/setup/values.yaml (hunk 1 above)
Comment (junior) [reply deploy/complete/helm-chart/setup/values.yaml:19]: I know. I need to change on the TF as well.

File: deploy/complete/terraform/mushop.tf (hunk 2 above)
Comment (junior) [reply deploy/complete/terraform/mushop.tf:24]: You mean this one: ```name  = "global.oadbAdminSecret"
    value = var.mushop_mock_mode_all ? "" : var.db_admin_name```

Comment (junior) [reply deploy/complete/terraform/mushop.tf:24]: ?

File: deploy/complete/terraform/schema.yaml (hunk 3 above)
Comment (junior) [reply deploy/complete/terraform/schema.yaml:10]: Thanks.

File: deploy/complete/terraform/mushop.tf (hunk 2 above)
Comment (mvandervliet) [reply deploy/complete/terraform/mushop.tf:24]: the three `global.oadb*`

File: deploy/complete/helm-chart/setup/values.yaml (hunk 1 above)
Comment (junior) [reply deploy/complete/helm-chart/setup/values.yaml]: Fixed

File: deploy/complete/terraform/schema.yaml (hunk 3 above)
Comment (junior) [reply deploy/complete/terraform/schema.yaml]: thanks @mvandervliet for the svg. Converted to data uri base64 and updated to the schema. The reduction was not much, but was something.
image

File: deploy/complete/terraform/mushop.tf (hunk 2 above)
Comment (junior) [reply deploy/complete/terraform/mushop.tf:24]: removed unnecessary fear conditions.

File: deploy/complete/terraform/mushop.tf [hunk 4]
```diff
@@ -7,7 +7,7 @@ resource "kubernetes_namespace" "mushop_namespace" {
   metadata {
     name = "mushop"
   }
-  depends_on = [oci_containerengine_node_pool.oke-mushop_node_pool]
+  depends_on = [oci_containerengine_node_pool.oke_mushop_node_pool]
```
Comment (jeevanjoseph) [inline deploy/complete/terraform/mushop.tf:10]: does the provider wait for the node to be in the AVAILABLE state ?

File: deploy/complete/terraform/schema.yaml [hunk 5]
```diff
@@ -119,7 +133,6 @@ variables:
   num_pool_workers:
     type: integer
     title: "Number of Worker Nodes"
-    default: 1
```
Comment (jeevanjoseph) [inline deploy/complete/terraform/schema.yaml:122]: Why remove the defaults ?

File: deploy/complete/terraform/mushop.tf (hunk 4 above)
Comment (junior) [reply deploy/complete/terraform/mushop.tf:10]: No, and is not needed. If the cluster is available, everything get scheduled to provision as soon is available.

oke terraform node_pool resource does not have any status control. This does not affects the solution deployment, as measures have been applied to have everything gracefully.

File: deploy/complete/terraform/schema.yaml (hunk 5 above)
Comment (junior) [reply deploy/complete/terraform/schema.yaml:122]: Only necessary defaults need to be on the schema. Defaults should come from variables always as possible.

**Pull request merged**